packages/client/src/store/action-creators/auth.ts

```diff
@@ -67,10 +71,31 @@ export const signOut = createAsyncThunk('user/signOut', async (_, thunkAPI) => {
   }
 });
 
+export const signInOAuth = createAsyncThunk(
+  'user/signInOAuth',
+  async (payload: { code: string; navigate: NavigateFunction }, thunkAPI) => {
+    const { code, navigate } = payload;
+    try {
+      const { data, error } = await signInWithProvider(code);
+      if (error) {
+        navigate('/sign-in');
+        return thunkAPI.rejectWithValue(error);
+      }
+      return data;
+    } catch (e: any) {
```

Comment:
```suggestion
    } catch (error: any) {
```

Используй более понятные имена переменных

Reply:
Поправлю в слудующей таске.